RubberduckTests/Grammar/ResolverTests.cs

```diff
@@ -2011,6 +2011,66 @@ End Sub
             }
         }
 
+        [Category("Grammar")]
+        [Category("Resolver")]
+        [Test]
+        public void RedimStmt_RedimVariableDeclarationCausesVariableDeclarationToBeCreated()
```

Comment:
I do not see any unit tests for `Redim`s with `As Long`. Those probably need to be covered. Furthermore, I think there needs to be negative unit tests to verify that no new declarations are created where we have a `Dim` for a given variable that's `Redim'`d.

Reply:
Rather than "As Long" there are no tests for any AsType declarations